pkg/scale/varying_data_type.go

```diff
@@ -58,7 +59,7 @@ type VaryingDataType struct {
 func (vdt *VaryingDataType) Set(value VaryingDataTypeValue) (err error) {
 	_, ok := vdt.cache[value.Index()]
 	if !ok {
-		err = fmt.Errorf("unable to append VaryingDataTypeValue: %T, not in cache", value)
+		err = fmt.Errorf("%w: %v (%T), not in cache", ErrAddVaryingDataTypeValue, value, value)
```

Comment:
Shouldn't the error `ErrAddVaryingDataTypeValue` be more like `ErrVaryingDataTypeValueNotInCache`? And then the `, not in cache` context string can also be removed? 🤔